ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManager.java

```diff
@@ -73,6 +78,10 @@ public class EthProtocolManager implements ProtocolManager, MinedBlockObserver {
   private final List<PeerValidator> peerValidators;
   // The max size of messages (in bytes)
   private final int maxMessageSize;
+  private Optional<Difficulty> powTerminalDifficulty;
+  private final StampedLock powTerminalDifficultyLock = new StampedLock();
+  private Hash lastFinalized = Hash.ZERO;
+  private final AtomicLong numFinalizedSeen = new AtomicLong(0);
```

Comment:
I think we are leaking too much of the merge implementation details into protocol manager.  

How about either:
* extracting and encapsulate the merge specific behavior into an Optional "PeerFilterAction" 
* or subclassing a merge-specific EthProtocolManager